src/features/farming/hud/components/DEV_BurnLandButton.tsx

```diff
@@ -0,0 +1,218 @@
+import React, { useContext, useEffect, useState } from "react";
+import { useActor } from "@xstate/react";
+import Web3 from "web3";
+import { AbiItem } from "web3-utils";
+
+import { Button } from "components/ui/Button";
+import { Context } from "features/auth/lib/Provider";
+
+const LAND_EXPANSION_ADDRESS = "0x1bAF4b9b954eD537c355469A47A995e3E4e31B48";
+
+const BALACE_OF_ABI: AbiItem = {
+  name: "balanceOf",
+  type: "function",
+  inputs: [
+    {
+      type: "address",
+      name: "owner",
+    },
+  ],
+};
+
+const TOKEN_OF_OWNER_BY_INDEX_ABI: AbiItem = {
+  name: "tokenOfOwnerByIndex",
+  type: "function",
+  inputs: [
+    {
+      type: "address",
+      name: "owner",
+    },
+    {
+      type: "uint256",
+      name: "index",
+    },
+  ],
+};
+
+const GAME_BURN_ABI: AbiItem = {
+  name: "gameBurn",
+  type: "function",
+  inputs: [
+    {
+      type: "uint256",
+      name: "tokenId",
+    },
+  ],
+};
+
+const loadBalance = async (web3: Web3, address: string) => {
+  const balanceOf = web3.eth.abi.encodeFunctionCall(BALACE_OF_ABI, [address]);
+
+  const balance = await web3.eth.call({
+    to: LAND_EXPANSION_ADDRESS,
+    data: balanceOf,
+  });
+
+  return parseInt(balance);
+};
+
+const loadTokenIds = async (web3: Web3, address: string, balance: number) => {
+  const tokenOfOwnerByIndex = (index: number) =>
+    web3.eth.abi.encodeFunctionCall(TOKEN_OF_OWNER_BY_INDEX_ABI, [
+      address,
+      String(index),
+    ]);
+
+  const tokenIds = await Promise.all(
+    Array.from({ length: balance }, (_, i) =>
+      web3.eth.call({
+        to: LAND_EXPANSION_ADDRESS,
+        data: tokenOfOwnerByIndex(i),
+      })
+    )
+  );
+
+  return tokenIds.map(Number);
+};
+
+const loadExpansions = async (web3: Web3, address: string) => {
+  const balance = await loadBalance(web3, address);
+  const tokenIds = await loadTokenIds(web3, address, balance);
+
+  return tokenIds;
+};
+
+const burnTokens = async (web3: Web3, tokens: Record<number, boolean>) => {
+  const account = (await web3.eth.getAccounts())[0];
+
+  const batch = new web3.BatchRequest();
+
+  const checkedTokens = Object.entries(tokens).filter(
+    ([id, checked]) => checked
+  );
+
+  let resolved = 0;
+
+  await new Promise((reject, resolve) => {
+    checkedTokens.forEach(([id]) =>
+      batch.add(
+        (web3.eth.sendTransaction as any).request(
+          {
+            from: account,
+            to: LAND_EXPANSION_ADDRESS,
+            data: web3.eth.abi.encodeFunctionCall(GAME_BURN_ABI, [id]),
+          },
+          () => ++resolved === checkedTokens.length && resolve()
```

Comment:
❤️‍🔥

Reply:
Lol I had a bit of fun because this is test code 🤣